Tests/Realm.Tests/Database/RemoveTests.cs

```diff
@@ -153,6 +153,36 @@ public void RemoveAllObjectsShouldClearTheDatabase()
             Assert.That(_realm.All<AllTypesObject>().Count(), Is.EqualTo(0));
         }
 
+        [Test]
+        public void RemoveAllObjectsShouldClearEntireDatabaseWithPartialSchema()
+        {
+            var c1 = new RealmConfiguration
+            {
+                Schema = new[] { typeof(Person), typeof(IntPrimaryKeyWithValueObject) }
+            };
+            var r1 = GetRealm(c1);
+            r1.Write(() =>
+            {
+                r1.Add(new Person());
+            });
+            r1.Dispose();
+
+            // Open with a subset of the schema
+            var c2 = new RealmConfiguration
+            {
+                Schema = new[] { typeof(IntPrimaryKeyWithValueObject) }
+            };
+            var r2 = GetRealm(c2);
+            r2.Write(() =>
+            {
+                r2.RemoveAll();
+            });
+
+            // Reopen with the complete schema
+            r1 = Realm.GetInstance(c1);
```

Comment:
So, interestingly enough. My attempt at just using `read_group()` without getting a full schema as @nirinchev suggested and it did allow me to pass the first test but throws the following here:
```
Realms.Tests.Database.RemoveTests.RemoveAllObjectsShouldClearEntireDatabaseWithPartialSchema:
    Outcome: Failed
    Error Message:
    Realms.Exceptions.RealmException : Object was deleted
    Stack Trace:
       at Realms.NativeException.ThrowIfNecessary(Func`2 overrider) in /Users/gagik.amaryan/Documents/realm-dotnet/Realm/Realm/Native/NativeException.cs:line 88
   at Realms.SharedRealmHandle.Open(Configuration configuration, RealmSchema schema, Byte[] encryptionKey) in /Users/gagik.amaryan/Documents/realm-dotnet/Realm/Realm/Handles/SharedRealmHandle.cs:line 398
   at Realms.RealmConfiguration.CreateHandle(RealmSchema schema) in /Users/gagik.amaryan/Documents/realm-dotnet/Realm/Realm/Configurations/RealmConfiguration.cs:line 142
   at Realms.RealmConfigurationBase.CreateRealm() in /Users/gagik.amaryan/Documents/realm-dotnet/Realm/Realm/Configurations/RealmConfigurationBase.cs:line 211
   at Realms.Realm.GetInstance(RealmConfigurationBase config) in /Users/gagik.amaryan/Documents/realm-dotnet/Realm/Realm/Realm.cs:line 104
   at Realms.Tests.Database.RemoveTests.RemoveAllObjectsShouldClearEntireDatabaseWithPartialSchema() in /Users/gagik.amaryan/Documents/realm-dotnet/Tests/Realm.Tests/Database/RemoveTests.cs:line 183
```
I am assuming it deletes more than objects with this call. What kind of "tables" are there?

Reply:
So... There should be also some tables that are used by realm for bookkeeping, so it would be good to ask someone from core how to recognise those

Reply:
You could also try debugging this and seeing which tables get cleared.